libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -1366,19 +1371,23 @@ double WaveClip::GetPlayEndTime() const
 {
     const auto numSamples = GetNumSamples();
     double maxLen = mSequenceOffset +
-                    ((numSamples + GetAppendBufferLen()).as_double()) *
-                       GetStretchRatio() / mRate -
-                    mTrimRight;
+                    ((numSamples + GetAppendBufferLen()).as_double() *
+                       GetStretchRatio() - floor(mTrimRight * mRate + 0.5)) / mRate;
     // JS: calculated value is not the length;
     // it is a maximum value and can be negative; no clipping to 0
-    return SnapToTrackSample(maxLen);
+    return maxLen;
 }
 
 double WaveClip::GetPlayDuration() const
 {
    return GetPlayEndTime() - GetPlayStartTime();
 }
 
+bool WaveClip::IsEmpty() const
+{
+   return GetPlayDuration() < 0.5 / mRate * mClipStretchRatio;
```

Comment:
Are you sure this shouldn't be `GetStretchRatio()` ? `IsEmpty()` should return the same value independently on tempo change, meanwhile `GetPlayDuration()` would.